spec/models/outlook/api_client_spec.rb

```diff
@@ -38,6 +32,24 @@
       transactionId: "agents_rdv-#{agents_rdv.id}",
     }
   end
+  let(:motif) { create(:motif, name: "Super Motif", location_type: :phone) }
+  let(:agent) { create(:agent, microsoft_graph_token: "token", refresh_microsoft_graph_token: "refresh_token") }
+  let(:user) { create(:user, email: "user@example.fr", first_name: "First", last_name: "Last", organisations: [organisation]) }
+  let(:rdv) { create(:rdv, users: [user], motif: motif, organisation: organisation, starts_at: Time.zone.parse("2023-01-01 11h00"), duration_in_min: 30, agents: [agent]) }
+  let(:agents_rdv) { rdv.agents_rdvs.first }
+
+  around do |example|
+    client_id = ENV.delete("AZURE_APPLICATION_CLIENT_ID")
+    ENV["AZURE_APPLICATION_CLIENT_ID"] = "fake_client_id"
+
+    client_secret = ENV.delete("AZURE_APPLICATION_CLIENT_SECRET")
+    ENV["AZURE_APPLICATION_CLIENT_SECRET"] = "fake_client_secret"
+
+    example.run
+
+    ENV["AZURE_APPLICATION_CLIENT_ID"] = client_id
+    ENV["AZURE_APPLICATION_CLIENT_SECRET"] = client_secret
+  end
```

Comment:
Il y a une version un peu plus précise, thread safe et fragile :smile: 

```suggestion
  before do
    allow(ENV).to receive(:fetch).with("AZURE_APPLICATION_CLIENT_ID", nil).and_return("fake_client_id")
    allow(ENV).to receive(:fetch).with("AZURE_APPLICATION_CLIENT_SECRET", nil).and_return("fake_client_secret")
  end
```

Ta version me va bien aussi, comme tu préfères. :wink: